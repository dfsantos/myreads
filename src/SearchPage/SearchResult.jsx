import React from 'react';
import PropTypes from 'prop-types';

import Book from '../common/Book';

const propTypes = {
  books: PropTypes.array,
  onCategorizeBook: PropTypes.func.isRequired,
};

const defaultProps = {
  books: [],
};

function SearchResult({ books, onCategorizeBook }) {
  return (
    <div className="search-books-results">
      <ol className="books-grid">
        {books.map(book => (
          <li key={book.id}>
            <Book title={book.title} onChangeBookCategory={onCategorizeBook(book)} />
          </li>
        ))}
      </ol>
    </div>
  );
}

SearchResult.propTypes = propTypes;
SearchResult.defaultProps = defaultProps;

export default SearchResult;
