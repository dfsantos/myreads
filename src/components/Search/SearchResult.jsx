import React from 'react';
import PropTypes from 'prop-types';

import Book from '../Book';

const propTypes = {
  searchQuery: PropTypes.string.isRequired,
  books: PropTypes.array,
  onCategorizeBook: PropTypes.func.isRequired,
};

const defaultProps = {
  books: [],
};

function SearchResult({ searchQuery, books, onCategorizeBook }) {
  const emptyResult = searchQuery.length > 0 && books.length === 0;
  return (
    <div className="search-books-results">
      {emptyResult && (
        <div className="empty-search-result">
          <span>Your search did not match any books.</span>
        </div>
      )}
      {!emptyResult && (
        <div className="bookshelf-books">
          <ol className="books-grid">
            {books.map(book => (
              <li key={book.id}>
                <Book
                  title={book.title}
                  subtitle={book.subtitle}
                  coverLink={book.imageLinks.thumbnail}
                  authors={book.authors}
                  description={book.description}
                  onChangeBookCategory={onCategorizeBook(book)}
                />
              </li>
            ))}
          </ol>
        </div>
      )}
    </div>
  );
}

SearchResult.propTypes = propTypes;
SearchResult.defaultProps = defaultProps;

export default SearchResult;
