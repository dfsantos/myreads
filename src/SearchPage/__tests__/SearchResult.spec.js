import React from 'react';
import toJson from 'enzyme-to-json';

import SearchResult from '../SearchResult';

const onCategorizeBookListener = jest.fn(() => () => {});

it('should render without errors', () => {
  shallow(<SearchResult onCategorizeBook={onCategorizeBookListener} />);
});

it('empty data snapshot', () => {
  const wrapper = shallow(<SearchResult onCategorizeBook={onCategorizeBookListener} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('not empty data snapshot', () => {
  const books = [{ id: 1, title: 'Book A' }, { id: 2, title: 'Book B' }];
  const wrapper = shallow(
    <SearchResult books={books} onCategorizeBook={onCategorizeBookListener} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('onCategorizeBookListener should be called with book as parameter', () => {
  const book = { id: 1, title: 'Book A' };
  const wrapper = shallow(
    <SearchResult books={[book]} onCategorizeBook={onCategorizeBookListener} />
  );
  expect(onCategorizeBookListener).toHaveBeenCalledWith(book);
});
