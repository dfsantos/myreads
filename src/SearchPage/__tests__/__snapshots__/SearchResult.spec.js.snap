// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty data snapshot 1`] = `
<div
  className="search-books-results"
>
  <ol
    className="books-grid"
  />
</div>
`;

exports[`not empty data snapshot 1`] = `
<div
  className="search-books-results"
>
  <ol
    className="books-grid"
  >
    <li
      key="1"
    >
      <Book
        onChangeBookCategory={[Function]}
        title="Book A"
      />
    </li>
    <li
      key="2"
    >
      <Book
        onChangeBookCategory={[Function]}
        title="Book B"
      />
    </li>
  </ol>
</div>
`;
